src/server.ts

```diff
@@ -0,0 +1,29 @@
+import Fastify from "fastify";
+import cors from "@fastify/cors";
+import projects from "./controllers/projects";
+
+const initializeServer = () => {
+  const server = Fastify({
+    logger: true,
+  });
+
+  server.register(cors, {
+    origin: "*",
+  });
+
+  server.get("/projects", projects);
+
+  server.listen(5030, (err, address) => {
+    if (err) {
+      console.error(err);
+
+      process.exit(1);
+    }
+
+    console.log(`Server listening at ${address}`);
+  });
+
+  return server;
+};
+
+export { initializeServer };
```

Comment:
By the nature of Fastify's test infrastructure. It was forcing me to delegate where server gets initialized and the app itself does. The reason for it is; the function `.inject` which is used to mock up the server's request needed to be called from the original fastify instance, means that with the previous implementation, which was initializing DB and server and calling them in one place didn't allow me to access this instance from the testing side. 
I hope you like this segregation... If not, I'd love to see other options you can suggest me. I'd love to hear and see! 🙌 

context: https://www.fastify.io/docs/latest/Guides/Testing